src/utils/combineMiddlewares.ts

```diff
@@ -0,0 +1,53 @@
+import {Middleware} from 'redux'
+import {
+    CustomMiddlewarePriority,
+    CustomMiddlewares,
+    CoreMiddlewareDescriptor
+} from '../interfaces/customMiddlewares'
+
+export function combineMiddlewares(
+    coreMiddlewares: CoreMiddlewareDescriptor,
+    customMiddlewares: CustomMiddlewares = null,
+) {
+    if (!customMiddlewares) {
+        return Object.values(coreMiddlewares)
+    }
+    const coreMiddlewaresKeys = Object.keys(coreMiddlewares) as Array<keyof typeof coreMiddlewares>
+
+    // get custom middlewares should be placed before core middlewares
+    const customMiddlewaresBefore: Middleware[] = []
+    const customMiddlewaresAfter: Middleware[] = []
+    Object.entries(customMiddlewares)
+    .filter(([cMKey, customMiddleware]) => !coreMiddlewaresKeys.includes(cMKey))
+    .forEach(([cMKey, customMiddleware]) => {
+        if (customMiddleware?.priority === CustomMiddlewarePriority.BEFORE) {
+            customMiddlewaresBefore.push(customMiddleware.implementation)
+            return
+        }
+        if (customMiddleware?.priority === CustomMiddlewarePriority.AFTER) {
+            customMiddlewaresAfter.push(customMiddleware.implementation)
+        }
+    })
+
+
+    // if there are present customMiddlewaresBefore then insert them before core middlewares
+    const resultMiddlewares: Middleware[] = customMiddlewaresBefore.length ? customMiddlewaresBefore : []
```

Comment:
It is almost identical to 
```ts
 const resultMiddlewares: Middleware[] = customMiddlewaresBefore
```
The only difference I can think of is that `[]` will give you are new array, but since you mutate original array in true-branch anyway It does not matter.

Reply:
@Dergash I have not got what 'original array' you are talking about.
This is an initialisation of `resultMiddlewares`. So it is either `customMiddlewaresBefore` or a new array.
But anyway, changed to:
```typescript
const resultMiddlewares: Middleware[] = customMiddlewaresBefore
```